cmd/firectl/main.go

```diff
@@ -188,23 +195,36 @@ func main() {
 		log.Fatalf("Invalid vsock specification: %s", err)
 	}
 
+	if len(opts.FcFifoLogFile) != 0 {
+		if len(opts.FcLogFifo) == 0 {
+			opts.FcLogFifo = filepath.Join(os.TempDir(), fmt.Sprintf("fifo_%s", randBytes(16)))
```

Comment:
Constructing a file path out of a pure pseudorandom byte sequence is unsafe and prone to failure or unexpected behavior.

Consider the case where a byte corresponds to a non-printing character such as a carriage return or a NULL character. Or where a byte corresponds to a character with special meaning to the filesystem, such as `0x2f`.

You might want to check out what [ioutil.TempFile()](https://golang.org/src/io/ioutil/tempfile.go#L39) is doing. Or use [ioutil.TempDir()](https://godoc.org/io/ioutil#TempDir) to create a temporary directory, then create the fifo in it with an arbitrary valid filename (can be a constant). After you open the fifo for reading, you can unlink the file and remove the temp directory.

Reply:
Yea, that's a good point with the `NULL` character and special characters in general. I went ahead and went with the `ioutil.TempDir` approach since that seemed simplest. I also changed the configuration parameter of `FifoLogFile` to `FifoLogWriter` which now gives users the choice on how they want to handle the fifo.